reporter.go

```diff
@@ -164,50 +164,36 @@ const (
 )
 
 type remoteReporter struct {
-	ReporterOptions
+	reporterOptions
 	sender       transport.Transport
 	queue        chan *zipkincore.Span
 	queueLength  int64 // signed because metric's gauge is signed
 	queueDrained sync.WaitGroup
 	flushSignal  chan *sync.WaitGroup
 }
 
-// ReporterOptions control behavior of the reporter
-type ReporterOptions struct {
-	// QueueSize is the size of internal queue where reported spans are stored before they are processed in the background
-	QueueSize int
-	// BufferFlushInterval is how often the buffer is force-flushed, even if it's not full
-	BufferFlushInterval time.Duration
-	// Logger is used to log errors of span submissions
-	Logger Logger
-	// Metrics is used to record runtime stats
-	Metrics *Metrics
-}
-
 // NewRemoteReporter creates a new reporter that sends spans out of process by means of Sender
-func NewRemoteReporter(sender transport.Transport, options *ReporterOptions) Reporter {
-	if options == nil {
-		options = &ReporterOptions{}
+func NewRemoteReporter(sender transport.Transport, options ...ReporterOption) Reporter {
+	reporter := &remoteReporter{
```

Comment:
this is a bit unintuitive, forces you to assign values to the struct in different places. I would rather keep the `options` var (rename param to `opts`), do all the defaults on it, and then at the end create the reporter struct with all the fields populated at once.